kernel: thread: move thread testcases of the tests/kernel/thread to new ZTEST api

Zhaoningx opened the pull request:
Move thread testcases of the tests/kernel/thread to new ZTEST api.

Commit:
kernel: thread: move thread dynamic cases to new ztest

Move thread dynamic testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move no-multithreading testcase to new ztest

Move thread no-multithreading testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move thread api testcase to new ztest

Move thread apis testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move thread error case to new ztest

Move thread error testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move thread init testcase to new ztest

Do some changes to make sure the testcases are independent.
Move thread init testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move thread stack testcase to new ztest

Move thread stack testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

Commit:
kernel: thread: move thread tls testcase to new ztest

Move thread tls testcases to new ztest.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

dcpleung commented on arch/x86/core/fatal.c:
```diff
@@ -28,6 +28,7 @@ FUNC_NORETURN void arch_system_halt(unsigned int reason)
 	 * (often due to bad page tables or other CPU structures) it will
 	 * terminate with 0 error code.
 	 */
+	printk("arch_system_halt in X86\n");
```
Please remove in this commit.

Zhaoningx commented:
Hi @andyross @dcpleung, I have addressed the comment and done some changes for thread init test cases, could you help to review this PR again? Thanks.

By the way, I don't know what should I do for the CI building failure, have you seen this failure log? Thanks.

dcpleung commented:
Errr.. the error said to increase `CONFIG_MAX_THREAD_BYTES`.

enjiamai commented:
I saw another failure like this, anyone knows how to solve it? 

```
INFO    -  74/406 degu_evk                  tests/kernel/threads/thread_apis/kernel.threads.apis FAILED Build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/degu_evk/tests/kernel/threads/thread_apis/kernel.threads.apis/build.log

.....

-- Found assembler: /opt/toolchains/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:22 (message):
  CONFIG_USB_DEVICE_VID has default value 0x2FE3.

  This value is only for testing and MUST be configured for USB products.


CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:28 (message):
  CONFIG_USB_DEVICE_PID has default value 0x100.

  This value is only for testing and MUST be configured for USB products.
```

Zhaoningx commented:
> Errr.. the error said to increase `CONFIG_MAX_THREAD_BYTES`.

Hi @dcpleung, thanks for your help, I have updated the PR.

Zhaoningx commented:
> I saw another failure like this, anyone knows how to solve it?
> 
> ```
> INFO    -  74/406 degu_evk                  tests/kernel/threads/thread_apis/kernel.threads.apis FAILED Build failure (build)
> INFO    - /__w/zephyr/zephyr/twister-out/degu_evk/tests/kernel/threads/thread_apis/kernel.threads.apis/build.log
> 
> .....
> 
> -- Found assembler: /opt/toolchains/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
> CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:22 (message):
>   CONFIG_USB_DEVICE_VID has default value 0x2FE3.
> 
>   This value is only for testing and MUST be configured for USB products.
> 
> 
> CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:28 (message):
>   CONFIG_USB_DEVICE_PID has default value 0x100.
> 
>   This value is only for testing and MUST be configured for USB products.
> ```

Hi @enjiamai, it seems the CI failure has been addressed. Thanks.
